src/west/configuration.py

```diff
@@ -173,26 +186,36 @@ def __init__(self, topdir: PathType | None = None):
         :param topdir: workspace location; may be None
         '''
 
-        local_path = _location(ConfigFile.LOCAL, topdir=topdir, find_local=False) or None
+        self._local_paths = self.get_location_paths(ConfigFile.LOCAL, topdir, find_local=False)
+        self._global_paths = self.get_location_paths(ConfigFile.GLOBAL, topdir)
+        self._system_paths = self.get_location_paths(ConfigFile.SYSTEM, topdir)
 
-        self._system_path = Path(_location(ConfigFile.SYSTEM, topdir=topdir))
-        self._global_path = Path(_location(ConfigFile.GLOBAL, topdir=topdir))
-        self._local_path = Path(local_path) if local_path is not None else None
+        self._system = _InternalCF.from_paths(self._system_paths)
+        self._global = _InternalCF.from_paths(self._global_paths)
+        self._local = _InternalCF.from_paths(self._local_paths)
 
-        self._system = _InternalCF.from_path(self._system_path)
-        self._global = _InternalCF.from_path(self._global_path)
-        self._local = _InternalCF.from_path(self._local_path)
-
-    def get_paths(self, location: ConfigFile = ConfigFile.ALL) -> list[Path]:
+    def get_search_paths(self, location: ConfigFile = ConfigFile.ALL) -> list[Path]:
         ret = []
-        if self._system and location in [ConfigFile.SYSTEM, ConfigFile.ALL]:
-            ret.append(self._system.path)
-        if self._global and location in [ConfigFile.GLOBAL, ConfigFile.ALL]:
-            ret.append(self._global.path)
-        if self._local and location in [ConfigFile.LOCAL, ConfigFile.ALL]:
-            ret.append(self._local.path)
+        if location in [ConfigFile.SYSTEM, ConfigFile.ALL]:
+            ret.extend(self._system_paths)
+        if location in [ConfigFile.GLOBAL, ConfigFile.ALL]:
+            ret.extend(self._global_paths)
+        if location in [ConfigFile.LOCAL, ConfigFile.ALL]:
+            ret.extend(self._local_paths)
         return ret
 
+    def get_existing_paths(self, location: ConfigFile = ConfigFile.ALL) -> list[Path]:
+        paths = self.get_search_paths(location)
+        return [p for p in paths if p.exists()]
```

Comment:
The `_InternalCF` objects have already filtered this, why not simply concatenate their `.paths` fields`? That keeps the filter in a single place and is a good reminder of what `_InternalCF` are.